src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -0,0 +1,474 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
```

Comment:
This class is essentially being ported as-is with changes just to recompile against internal vs. extension APIs. May need refresh since we took some small changes in the extension.